### PR TITLE
#1228 support multi roots in language service

### PR DESCRIPTION
**Pull request by DonJayamanne**
multi root support for language services such as intellisense, signature, code navigation, etc

### CODE REVIEW

**Comment by DonJayamanne** [comment]
wrong branch

File: src/client/providers/jediProxy.ts [hunk 1]
```diff
@@ -334,250 +381,201 @@ function spawnProcess(dir: string) {
...
-    if (cmd.command === CommandType.Symbols) {
-        delete payload.column;
-        delete payload.line;
-    }
-
-    return payload;
-}
+        catch (ex) {
+            console.error(ex);
+            //If 'This socket is closed.' that means process didn't start at all (at least not properly)
```
Comment (brettcannon) [inline src/client/providers/jediProxy.ts]: Not going to block approval, but missing a period at the end (and on most other comments).

File: src/client/providers/jediProxy.ts [hunk 2]
```diff
@@ -334,250 +381,201 @@ function spawnProcess(dir: string) {
...
-        filePaths.push(Promise.resolve(PYTHONPATH.trim()));
+
+    private lastKnownPythonPath: string = null;
+    private additionalAutoCopletePaths: string[] = [];
+    private getPathFromPythonCommand(args: string[]): Promise<string> {
+        return execPythonFile(this.pythonSettings.pythonPath, args, this.workspacePath).then(stdout => {
+            if (stdout.length === 0) {
+                return "";
+            }
+            let lines = stdout.split(/\r?\n/g).filter(line => line.length > 0);
```
Comment (brettcannon) [inline src/client/providers/jediProxy.ts]: Is there an npm package to do appropriate line splitting? (To start using in the future.)

File: src/client/providers/jediProxy.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/providers/jediProxy.ts]: Thanks, for once we get over the multiroot, please do block approval. I want code when adding multi root support.

File: src/client/providers/jediProxy.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/providers/jediProxy.ts]: There is one, but not popular enough. I guess the problem is this is a very simple piece of code. Can create a function for this though, but not sure its worth it.

Comment (brettcannon) [reply src/client/providers/jediProxy.ts]: Nah, it's fine. I just always assume anything that has been written in JavaScript more than once has a package on npm. 😉

**Pull request merged**